### PR TITLE
fix(modules): Exit gracefully if source map file doesn't exist

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description
If a user supplies a wrong path or the assumed path doesn't exist it would crash the build.

## :bulb: Motivation and Context
fixes: https://github.com/getsentry/sentry-react-native/issues/2654

## :pencil: Checklist

-  [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Comment by marandaneto** [comment]
@krystofwoldrich can we add tests to avoid regression?

**Comment by krystofwoldrich** [comment]
@marandaneto Yes, I will move the validation logic to the class so it's easy to test.

File: src/js/tools/ModulesCollector.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
-import { logger } from '@sentry/utils';
```
Comment (marandaneto) [inline src/js/tools/ModulesCollector.ts:1]: M: Why aren't using the `logger` anymore?

File: src/js/tools/ModulesCollector.ts [hunk 2]
```diff
@@ -1,7 +1,9 @@
-/* eslint-disable no-console */
+import { logger } from '@sentry/utils';
 import { existsSync, mkdirSync, readFileSync, writeFileSync } from 'fs';
 import { posix, sep } from 'path';
 
+logger.enable();
```
Comment (marandaneto) [inline src/js/tools/ModulesCollector.ts:5]: M: Is this needed? I don't see this anywhere else.

**Review by marandaneto — APPROVED** [review]
Left a last [comment](https://github.com/getsentry/sentry-react-native/pull/2655/commits/25b064122e55584b85d151c4ea8afd18a8961312#r1036798853) otherwise LGTM

File: src/js/tools/ModulesCollector.ts (hunk 2 above)
Comment (krystofwoldrich) [reply src/js/tools/ModulesCollector.ts:5]: For RN SDK it's hidden in [`initAndBind`](https://github.com/getsentry/sentry-javascript/blob/8570f0d29567e5f7cebf9a18f82e019191f2ebee/packages/core/src/sdk.ts#L22)
But as this run as a standalone script, I needed to enable the logger.

**Pull request merged**